default/python/AI/ResourcesAI.py

```diff
@@ -317,10 +360,11 @@ def assess_protection_focus(pinfo):
     this_planet = pinfo.planet
     aistate = get_aistate()
     sys_status = aistate.systemStatus.get(this_planet.systemID, {})
-    threat_from_supply = (0.25 * aistate.empire_standard_enemy_rating *
-                          min(2, len(sys_status.get('enemies_nearly_supplied', []))))
+    threat_from_supply = (
+        0.25 * aistate.empire_standard_enemy_rating * min(2, len(sys_status.get("enemies_nearly_supplied", [])))
+    )
     debug("%s has regional+supply threat of %.1f", this_planet, threat_from_supply)
-    regional_threat = sys_status.get('regional_threat', 0) + threat_from_supply
+    regional_threat = sys_status.get("regional_threat", 0) + threat_from_supply
     if not regional_threat:  # no need for protection
         if pinfo.current_focus == PROTECTION:
             debug("Advising dropping Protection Focus at %s due to no regional threat", this_planet)
```

Comment:
after looking at the code for almost half an hour, i came to this point here, and i believe this may be a candidate cause for why the volp-uglush and the setinon and the nightsiders native planets do not default to defense focus but instead industry. shouldnt there be some sort of checking here if the local species like or dislike that focus? i'll look some more at the code, maybe i'm completely wrong ...

Reply:
see also https://github.com/freeorion/freeorion/pull/3665#issuecomment-1013861481

Reply:
> after looking at the code for almost half an hour

You are quite fast in reading tons of lines.  This code is overcomplicated. :)

This is code for the AI player to handle its empire. That means that this planet is already conquered by the empire.  Changes here will not address your problem that natives are not defensive enough.